docs/api_reference/versioning.md

```diff
@@ -0,0 +1,69 @@
+# Versioning Strategy for APIs
+
+## Current approach
+In the current workflow, we version Jellyfish together with OpenAPI and protobuf.
+The same version number is applied to SDKs and we enforce using the same Jellyfish and SDK versions.
+
+This is great from a developer perspective, but causes issues from
+the user perspective.
+The main problem is the necessity to update both Jellyfish and SDK
+at the same moment.
+
+### Pros
+* easy to introduce changes (no backward compatibility required)
+* easy to test - only test with the current version
+
+### Cons
+* no backward compatibility
+
+
+## Proposed Approach
+Introducing breaking changes gradually.
+
+This scenario proposes introducing breaking changes gradually,
+first by making the old features deprecated, and removing them
+later, giving users some time to adapt to the changes.
+
+This allows for the progressive introduction of new versions of Jellyfish,
+e.g. first by updating the Jellyfish Server and then the SDK.
+This also allows for working with two system versions at the same time
+(SDK connecting to different Jellyfish versions),
+until everything is upgraded.
+
+### Example
+Let's say that we introduce backward-compatible changes between Jellyfish
+`0.2.0` versions and `0.4.0` versions. In the process, we marked
+some fields as deprecated and then remove them in Jellyfish `0.5.0`,
+while preserving compatibility with JF `0.3.0` and `0.4.0`.
+Similarly, in the version `0.6.0` we will likely break compatibility
+with `0.3.0` but remain compatible with `0.4.0` and `0.5.0`.
+
+In this scenario, we need to decide how to version the API.
+The best way would be to keep it the same as Jellyfish,
+since most of the time any change in REST or WebSocket causes
+changes in Jellyfish.
+
+The downside is that a change in REST API triggers a version change
+in WebSocket API and vice versa. This is a disadvantage if someone uses 
+solely either REST API or WebSocket API - they would have to
+update the version even if there was no change in the API they were using.
```

Comment:
I think we should keep OpenAPISpec separate from WS. I mean, it will look strange when we bump OpenAPISpec version just because we changed something in the WS API.

Do we have any examples we could take pattern from?

Reply:
[Jitsi, for example, for mobile SDKs, has one release note for both Android and iOS and simply writes `N/A` if nothing changes from the previous version](https://github.com/jitsi/jitsi-meet-release-notes/blob/master/CHANGELOG-MOBILE-SDKS.md).
Maintaining a separate additional table informing which version of OpenAPISpec is compatible with WS sounds like something that will be always outdated. IMO to keep versioning up-to-date must be stupidly simple e.g.: everything with version 0.2.0 should work together and that's all you don't have to think about anything else.

Reply:
I agree with @Rados13, I don't think the benefits of having two separate version for OpenAPISpec and WS are great enough to consider doing it.

We'd have to think about one more potential point of failure in resolving the Jellyfish-ecosystem dependency hell (which at the moment seems pretty tame, and I'd rather keep it that way)